google/cloud/pubsub/samples/samples.cc

```diff
@@ -221,6 +221,109 @@ void Subscribe(google::cloud::pubsub::Subscriber subscriber,
   (std::move(subscriber), std::move(subscription));
 }
 
+void CustomThreadPoolPublisher(std::vector<std::string> const& argv) {
+  namespace examples = ::google::cloud::testing_util;
+  if (argv.size() != 2 || (argv.size() == 1 && argv[0] == "--help")) {
+    throw examples::Usage{
+        "custom-thread-pool-publisher <project-id> <topic-id>"};
+  }
+  //! [custom-thread-pool-publisher]
+  namespace pubsub = google::cloud::pubsub;
+  using google::cloud::future;
+  using google::cloud::StatusOr;
+  [](std::string project_id, std::string topic_id) {
+    // Create our own completion queue to run the background activity, such as
+    // flushing the publisher.
+    google::cloud::CompletionQueue cq;
+    // Setup 1 or more of threads to service this completion queue, these must
+    // remain running until all the work is done.
+    std::vector<std::thread> tasks;
+    std::generate_n(std::back_inserter(tasks), 4, [&cq] {
+      return std::thread(&google::cloud::CompletionQueue::Run, &cq);
```

Comment:
We may not want to advocate for users taking addresses of our functions.  That is, use a lambda instead?

Reply:
Hmmm... Okay, I understand why.  Done.